src/vector_index.c

```diff
@@ -104,6 +131,37 @@ IndexIterator *NewVectorIterator(QueryEvalCtx *q, VectorQuery *vq, IndexIterator
       };
       return NewHybridVectorIterator(hParams);
     }
+    case VECSIM_QT_RANGE: {
```

Comment:
IMO, this should be in a different function since there is no real correlation between `KNN` and `RANGE`.

It can be done later.